Neos.ContentRepository.Core/Classes/Feature/ContentStreamHandling.php

```diff
@@ -0,0 +1,224 @@
+use Neos\ContentRepository\Core\SharedModel\Exception\ContentStreamIsClosed;
+use Neos\ContentRepository\Core\SharedModel\Exception\ContentStreamIsNotClosed;
+use Neos\ContentRepository\Core\SharedModel\Workspace\ContentStreamId;
+use Neos\ContentRepository\Core\SharedModel\Workspace\ContentStreamStatus;
+use Neos\EventStore\Model\EventStream\ExpectedVersion;
+
+trait ContentStreamHandling
+{
+    /**
+     * @param ContentStreamId $contentStreamId The id of the content stream to create
+     * @throws ContentStreamAlreadyExists
+     */
+    private function createContentStream(
+        ContentStreamId $contentStreamId,
+        CommandHandlingDependencies $commandHandlingDependencies,
+    ): EventsToPublish {
+        $this->requireContentStreamToNotExistYet($contentStreamId, $commandHandlingDependencies);
+        $streamName = ContentStreamEventStreamName::fromContentStreamId($contentStreamId)
+            ->getEventStreamName();
+
+        return new EventsToPublish(
+            $streamName,
+            Events::with(
+                new ContentStreamWasCreated(
+                    $contentStreamId,
+                )
+            ),
+            ExpectedVersion::NO_STREAM()
+        );
+    }
+
+    /**
+     * @param ContentStreamId $contentStreamId The id of the content stream to close
+     * @param CommandHandlingDependencies $commandHandlingDependencies
+     * @return EventsToPublish
+     */
+    private function closeContentStream(
+        ContentStreamId $contentStreamId,
+        CommandHandlingDependencies $commandHandlingDependencies,
+    ): EventsToPublish {
+        $this->requireContentStreamToExist($contentStreamId, $commandHandlingDependencies);
+        $expectedVersion = $this->getExpectedVersionOfContentStream($contentStreamId, $commandHandlingDependencies);
+        $this->requireContentStreamToNotBeClosed($contentStreamId, $commandHandlingDependencies);
+        $streamName = ContentStreamEventStreamName::fromContentStreamId($contentStreamId)->getEventStreamName();
+
+        return new EventsToPublish(
+            $streamName,
+            Events::with(
+                new ContentStreamWasClosed(
+                    $contentStreamId,
+                ),
+            ),
+            $expectedVersion
+        );
+    }
+
+    /**
+     * @param ContentStreamId $contentStreamId The id of the content stream to reopen
+     * @param ContentStreamStatus $previousState The state the content stream was in before closing and is to be reset to
+     */
+    private function reopenContentStream(
+        ContentStreamId $contentStreamId,
+        ContentStreamStatus $previousState,
+        CommandHandlingDependencies $commandHandlingDependencies,
+    ): EventsToPublish {
+        $this->requireContentStreamToExist($contentStreamId, $commandHandlingDependencies);
+        // todo dont we want to make sure this always succeeds? e.g. ANY?
```

Comment:
old code but new question:

shouldnt reopen always work????? eg any?

Reply:
I'll have a look

Reply:
Given that we use it ONLY in exception cases I think it makes sense to just ANY....

Reply:
jup see any https://github.com/neos/neos-development-collection/pull/5315/commits/858962a90ab37223b4804a3a02b963f28e18d049